cni/pkg/plugin/ambient.go

```diff
@@ -51,7 +51,7 @@ func checkAmbient(conf Config, ambientConfig ambient.AmbientConfigFile, podName,
 		return false, err
 	}
 
-	if ambientpod.HasLegacyLabel(pod.Labels) || ambientpod.HasLegacyLabel(ns.Labels) {
+	if ambientpod.HasLegacyLabel(ns.Labels) {
```

Comment:
L58 has checked `ShouldPodBeInMesh`

Reply:
does seem like it could be moved to an `else` statement after the check on L58, since `ShouldPodBeInMesh` does perform the `HasLegacyLabels` check.  The dedicated legacy labels error message is nice - but are we expecting a lot of users to hit this error scenario and need to know why their pod wasn't added under ambient?

Reply:
Having a label doesn't necessarily mean its injected. If we want improved error messages we should have ShouldPodBeInMesh do it imo